### ISSUE TITLE
Input validation for plot callbacks

### ISSUE THREAD

**Issue by ngoldbaum** [yt-project/yt#1839]
### Bug report

**Summary**

Many plot callbacks do not do proper input validation. This leads to confusing error messages or silently incorrect results. We should make use of  the input validation helper functions added in #1833 to the plot callbacks as well as the data objects.

Below is an example from a user bug report. It took me about 10 minutes of staring at the code to figure out where the error was coming from.

**Code for reproduction**

```python
import yt

ds = yt.load("IsolatedGalaxy/galaxy0030/galaxy0030")

val, loc = ds.find_max('density')
plt = yt.SlicePlot(ds, 'z', 'density)
plt.annotate_particles(ds, 10000*pc)
plt.save("test.png")
```

**Actual outcome**

```python traceback
Traceback (most recent call last):
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_window.py", line 1001, in run_callbacks
    callback(cbw)
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_modifications.py", line 64, in _check_geometry
    return func(self, plot)
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_modifications.py", line 1619, in __call__
    reg = self._get_region((x0,x1), (y0,y1), plot.data.axis, data)
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_modifications.py", line 1680, in _get_region
    LE[zax] = data.center[zax] - self.width*0.5
TypeError: unsupported operand type(s) for *: 'EnzoDataset' and 'float'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 14, in <module>
    plt.save("test.png")
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_container.py", line 95, in newfunc
    args[0]._setup_plots()
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_window.py", line 950, in _setup_plots
    self.run_callbacks()
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_window.py", line 1007, in run_callbacks
    sys.exc_info()[2])
  File "/usr/local/lib/python3.6/site-packages/six.py", line 692, in reraise
    raise value.with_traceback(tb)
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_window.py", line 1001, in run_callbacks
    callback(cbw)
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_modifications.py", line 64, in _check_geometry
    return func(self, plot)
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_modifications.py", line 1619, in __call__
    reg = self._get_region((x0,x1), (y0,y1), plot.data.axis, data)
  File "/Users/goldbaum/Documents/yt-git-fixes/yt/visualization/plot_modifications.py", line 1680, in _get_region
    LE[zax] = data.center[zax] - self.width*0.5
yt.utilities.exceptions.YTPlotCallbackError: annotate_particles callback failed with the following error: unsupported operand type(s) for *: 'EnzoDataset' and 'float'
```

**Expected outcome**

Plot gets saved to disk

**Version Information**

  * Operating System: Mac
  * Python Version: 3.6
  * yt version: yt master
  * Other Libraries (if applicable):

**Comment by git-abhishek** [comment]
I will work on this.

**Comment by ngoldbaum** [comment]
Remilestoned for 4.0 because this isn't a bug.

**Comment by neutrinoceros** [comment]
I think I missed this ticket when I opened (dup) #3945
However it's been resolved in #3957 and the patch was successfully released in yt 4.1 so I'll close
